pkg/api/applications/v2/api_test.go

```diff
@@ -197,39 +231,23 @@ func runTest(t *testing.T, td *apitest.ApplicationTestDefinition, appAPI applica
 						require.NoError(t, err, "failed to report trial")
 					}
 				})
-			}
-
-			err = appAPI.DeleteActivity(ctx, ai.URL)
-			require.NoError(t, err, "failed to acknowledge activity")
-		}
-	})
-
-	t.Run("Request Activity", func(t *testing.T) {
-		t.Parallel()
-		if !ok {
-			t.Skip("skipping activity request.")
-		}
-
-		md, err := appAPI.CheckEndpoint(ctx)
-		require.NoError(t, err, "failed to fetch the application list necessary for the feed URL")
+				t.Run("Acknowledge Run Activity", func(t *testing.T) {
+					err = appAPI.DeleteActivity(ctx, ai.URL)
+					require.NoError(t, err, "failed to acknowledge run activity")
+				})
 
-		sa := &applications.ScanActivity{
-			Scenario: scnMeta.Location(),
-		}
-		err = appAPI.CreateActivity(ctx, md.Link(api.RelationAlternate), applications.Activity{Scan: sa})
-		require.NoError(t, err, "failed to request scan")
+			}
 
-		ra := &applications.RunActivity{
-			Scenario: scnMeta.Location(),
+			// If we processed both activities, cancel the subscription early instead of waiting for the timeout
+			if okScan && okRun {
+				cancelSub()
+			}
 		}
-		err = appAPI.CreateActivity(ctx, md.Link(api.RelationAlternate), applications.Activity{Run: ra})
-		require.NoError(t, err, "failed to request run")
-		run.Add(1)
-	})
 
-	// Wait for all run activities to finish
-	run.Wait()
-	cancelSub()
+		// Make sure we witnessed both the scan and run activities for our scenario
+		assert.True(t, okScan, "never received the scan activity")
+		assert.True(t, okRun, "never received the run activity")
+	}) && ok
 
 	t.Run("Delete Application", func(t *testing.T) {
```

Comment:
For "delete" the skip condition is different because we always want clean up the application if the URL is present.

This is a case I prefer to leave the redundant assignment to help with refactoring: if anything is added or moved around, the assignment may not be redundant anymore.